src/components/table-hoc/TableHOCUtils.ts

```diff
@@ -1,13 +1,35 @@
 import * as _ from 'underscore';
 import {IReactVaporState} from '../../ReactVapor';
 import {IFilterState} from '../filterBox/FilterBoxReducers';
+import {IListBoxState} from '../listBox/ListBoxReducers';
 import {ITableWithSortState} from './reducers/TableWithSortReducers';
 
+export const TableHOCComponentType = {
+    Predicate: 'predicate',
+    Pagination: 'pagination',
+};
+
+const getComponentId = (tableId: string, type: string, componentId?: string) => {

```

Comment:
There are some use case with this method that are weird:

* `getComponentId(tableId, "predicate")`: There is no way to know that, in this case `componentId` is required.
* `getComponentId(tableId, "pagination", componentId)`: `componentId` would be unused.
* `getComponentId(tableId, "pagnation", componentId)`: I have a typo for the type, so it sneakily gets into the `default` path.

I think it should be split into different methods just to ensure that there is no incoherent parameters.

* `getPredicateId(tableId, componentId)`
* `getPaginationId(tableId)`

would be easier to use